libraries/botbuilder-dialogs/src/skillDialog.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * @module botbuilder-dialogs
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { BotFrameworkSkill } from 'botbuilder'; // bad import
+import { Activity, ActivityTypes, ConversationState, StatePropertyAccessor, TurnContext } from 'botbuilder-core';
+import { Dialog, DialogTurnResult } from './dialog';
+import { DialogContext } from './dialogContext';
+import { SkillDialogArgs } from './skillDialogArgs';
+import { SkillDialogOptions } from './skillDialogOptions';
+
+export class SkillDialog extends Dialog {
+    private readonly _activeSkillProperty: StatePropertyAccessor;
+    private readonly _conversationState: ConversationState; // Why is this restricted to ConversationState?
```

Comment:
Shouldn't this be `BotState`? 

https://github.com/microsoft/botbuilder-dotnet/blob/69112a39fa0aa7fb5cea6e506f41b36779c33681/libraries/Microsoft.Bot.Builder.Skills/Dialogs/SkillDialog.cs#L25

Reply:
Going to stay with ConversationState because we're saving ConversationState and there are stronger semantics around it (SkillDialog needs to save the DialogStack which is normally saved in ConversationState).